lib/_tls_wrap.js

```diff
@@ -383,13 +395,6 @@ TLSSocket.prototype._init = function(socket, wrap) {
   if (options.handshakeTimeout > 0)
     this.setTimeout(options.handshakeTimeout, this._handleTimeout);
 
-  // Socket already has some buffered data - emulate receiving it
-  if (socket && socket._readableState && socket._readableState.length) {
-    var buf;
-    while ((buf = socket.read()) !== null)
-      ssl.receive(buf);
-  }
-
```

Comment:
Unless I'm missing something, I need to defer the `ssl.receive(buf);` in order to fix the issue I'm trying to fix with this PR. If I defer that, I need to ALSO defer the `this.read(0)` in the constructor or it will start to read data off the socket before the existing data was read in. Please correct me if I'm wrong.